pkg/cli/sql.go

```diff
@@ -647,7 +658,7 @@ func (c *cliState) doStartLine(nextState cliStateEnum) cliStateEnum {
 	c.partialLines = c.partialLines[:0]
 	c.partialStmtsLen = 0
 
-	if isInteractive {
+	if c.ins != noLineEditor {
```

Comment:
nit: this read clearer before - could you add a bool for this with a descriptive name? like hasLineEditor.